arm-recoveryservices/2016-06-01/swagger/vaults.json

```diff
@@ -0,0 +1,763 @@
+      "description": "Operation properties.",
+      "type": "object",
+      "properties": {
+        "logSpecifications": {
+          "description": "List of log specifications of this operation.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientDiscoveryForLogSpecification"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "ClientDiscoveryResponse": {
+      "description": "List of available operations.",
+      "type": "object",
+      "properties": {
+        "Value": {
+          "description": "List of available operationss.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientDiscoveryValueForSingleApi"
+          },
+          "readOnly": true
+        },
+        "NextLink": {
+          "description": "NextLink",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ClientDiscoveryValueForSingleApi": {
+      "description": "Available operation details.",
+      "type": "object",
+      "properties": {
+        "Name": {
+          "description": "Name of the operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "Display": {
+          "$ref": "#/definitions/ClientDiscoveryDisplay",
+          "description": "Contains the localized display information for this particular operation.",
+          "readOnly": true
+        },
+        "Origin": {
+          "description": "The intended executor of the operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "Properties": {
+          "$ref": "#/definitions/ClientDiscoveryForProperties",
+          "description": "Properties of the operation.",
+          "readOnly": true,
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "Resource": {
+      "description": "ARM Resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource Id represents the complete path to the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name associated with the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type represents the complete path of the form Namespace/ResourceType/ResourceType/...",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "eTag": {
+          "description": "Optional ETag.",
+          "type": "string"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "NonTrackedResource": {
```

Comment:
NonTrackedResource cannot have the location property. Also if you wish to have tracked and proxy (non-tracked) resources do the following:
- Define a base class called ```Resource``` with ```id, name, tag readonly properties```
- Define a ```tracked resource``` with ```location``` as ```required``` property that allOfs on ```Resource```
- Define a ```proxy resource (non tracked resource)``` that allOfs on ```Resource``` and has any additional properties you wish to add